cli/src/klio_cli/commands/job/gke.py

```diff
@@ -0,0 +1,227 @@
+            # Strip off existing image tag if present
+            image_base = re.split(":", image_base)[0]
+            full_image = image_base + f":{image_tag}"
+            glom.assign(self._deployment_config, image_path, full_image)
+            log_messages.append(
+                f"Update deployment with image tag {image_tag}"
+            )
+        resp = self.kubernetes_client.patch_namespaced_deployment(
+            name=deployment_name,
+            namespace=namespace,
+            body=self.deployment_config,
+        )
+        log_messages.append(f"Update deployment with {resp.metadata.name}")
+        for message in log_messages:
+            logging.info(message)
+
+
+class RunPipelineGKE(GKECommandMixin, base.BaseDockerizedPipeline):
+    def __init__(
+        self, job_dir, klio_config, docker_runtime_config, run_job_config
+    ):
+        super().__init__(job_dir, klio_config, docker_runtime_config)
+        self.run_job_config = run_job_config
+
+    def _apply_image_to_deployment_config(self):
+        image_tag = self.docker_runtime_config.image_tag
+        if image_tag:
+            dep = self.deployment_config
+            image_path = "spec.template.spec.containers.0.image"
+            # TODO: If more than one image deployed,
+            #  we need to search for correct container
+            image_base = glom.glom(dep, image_path)
+            # Strip off existing image tag if any
+            image_base = re.split(":", image_base)[0]
+            full_image = f"{image_base}:{image_tag}"
+            glom.assign(self._deployment_config, image_path, full_image)
+
+    def _apply_deployment(self):
+        """Create a namespaced deploy if the deployment does not already exist.
+        If the namespaced deployment already exists then
+        `self.run_job_config.update` will determine if the
+        deployment will be updated or not.
+        """
+        dep = self.deployment_config
+        namespace = glom.glom(dep, "metadata.namespace")
+        deployment_name = glom.glom(dep, "metadata.name")
+        if not self._deployment_exists():
+            resp = self.kubernetes_client.create_namespaced_deployment(
+                body=dep, namespace=namespace
+            )
+            deployment_name = resp.metadata.name
+            logging.info(f"Deployment created for {deployment_name}")
+        else:
+            if self.run_job_config.update:
+                self._update_deployment()
+            else:
+                logging.warning(
+                    f"Cannot apply deployment for {deployment_name}. "
+                    f"To update an existing deployment, run "
+                    f"`klio job run --update`, or set `pipeline_options.update`"
+                    f" to `True` in the job's`klio-job.yaml` file. "
+                    f"Run `klio job stop` to scale a deployment down to 0. "
+                    f"Run `klio job delete` to delete a deployment entirely."
```

Comment:
FYI there's no need to add the `f` prefix for lines that don't have any variables (basically all but the first one). No need to change it here; but just wanted to surface it.